microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,276 @@
+        )
+
+        local_kselftest_results_path = PurePath(log_path) / "kselftest-results.txt"
+
+        # parse results from local_kselftest_results_path file
+        # read the file
+        with open(local_kselftest_results_path, encoding="utf8") as f:
+            result_output = f.read()
+            results = self._parse_results(result_output)
+
+        if not results:
+            raise LisaException("tests did not run, kselftest-results.txt is empty")
+
+        # assert that all tests passed
+        failed_tests = []
+        for result in results:
+            if result.status == TestStatus.FAILED:
+                failed_tests.append(result.name)
+
+            # create test result message
+            info: Dict[str, Any] = {}
+            info["information"] = {}
+            info["information"]["name"] = result.name
+            info["information"]["exit_value"] = result.exit_value
+            subtest_message = create_test_result_message(
+                SubTestMessage, test_result, environment, result.name, result.status
+            )
+
+            # notify subtest result
+            notifier.notify(subtest_message)
+
+        # assert that none of the tests failed
+        assert_that(
+            failed_tests, f"The following tests failed: {failed_tests}"
+        ).is_empty()
+
+        return results
+
+    def _parse_results(self, result: str) -> List[KselftestResult]:
+        parsed_result: List[KselftestResult] = []
+
+        lines_list = result.splitlines()
+
+        for line in lines_list:
+            line_str = ''.join(line)
+            matched = find_groups_in_lines(line_str, self._RESULT_KSELFTEST_OK_REGEX)
+
+            # get testcase data
+            if matched:
+                # example parsed_result: name='breakpoint_test_arm64',
+                # status=<TestStatus.FAILED: 4>, exit_value='1'
+                parsed_result.append(
+                    KselftestResult(
+                        name=matched[0]["name"],
+                        status=self._parse_status_to_test_status(matched),
+                        exit_value=self._parse_exit_val_to_test_status(matched),
+                    )
+                )
+                print("********")
+                print(matched[0]["name"])
+
+        return parsed_result
+
+    def _parse_status_to_test_status(self, result_list: List[Dict[str, str]]) -> TestStatus:
+        # example skip test log: "6 selftests: cgroup: test_stress.sh # SKIP"
+        # example failed test log: "52 selftests: net: veth.sh # exit=1"
+        # example passed test log: "2 selftests: breakpoints: breakpoint_test"
+        # example timeout test log: not ok 8 selftests: netfilter: nft_concat_range.sh # TIMEOUT 45 seconds
+
+        status = result_list[0]["status"]
+        exit_code = result_list[0]["info"].split("#")
+
+        # passed or skipped tests
+        if status == 'ok':
+            if len(exit_code) == 2:
+                if exit_code[1] == "SKIP":
```

Comment:
What should the test status be on the `else` block? It will return a `None` here.